CMakeLists.txt

```
@@ -54,6 +55,27 @@ if (SOCI_LTO)
     endif()
 endif()
 
+if (SOCI_VISIBILITY)
+    # Test whether visibility is supported
+    include(CheckCSourceCompiles)
+    check_c_source_compiles(
+        "
+            __attribute__ (( visibility(\"default\") )) int f1() { return 0; }
+            __attribute__ (( visibility(\"hidden\") ))  int f2() { return 1; }
+
+            int main(int argc, char* argv[]) { f1(); f2(); return 0; }
+        "
+        ENABLE_VISIBILITY_SUPPORT
```

```suggestion
        SOCI_ENABLE_VISIBILITY_SUPPORT
```
just to avoid any possible (even if unlikely) conflict with another library predefining this symbol when compiling SOCI headers.

---

Sure.